miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/api/ApiClient.kt

```diff
@@ -37,14 +37,22 @@ internal class ApiClient @VisibleForTesting constructor(
         hostAppId = rasAppId
     )
 
+    @Throws(MiniAppSdkException::class)
     suspend fun list(): List<MiniAppInfo> {
         val request = appInfoApi.list(hostAppId, hostAppVersionId)
         return requestExecutor.executeRequest(request)
     }
 
+    @Throws(MiniAppSdkException::class)
     suspend fun fetchInfo(appId: String): MiniAppInfo {
         val request = appInfoApi.fetchInfo(hostAppId, hostAppVersionId, appId)
-        return requestExecutor.executeRequest(request).first()
+        val info = requestExecutor.executeRequest(request)
+
+        if (info.isNotEmpty()) {
+            return info.first()
+        } else {
+            throw MiniAppSdkException("Server returned no info for the Mini App Id: $appId")
```

Comment:
Having it return the real error is good.
But in reality, this would only happen if there's some internal serval error.